### PR TITLE
Add sanity, linters, changelog, black and units workflows

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Add sanity, linters, changelog, black and units workflows

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: .github/workflows/black.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+name: black
```
Comment (GomathiselviS) [inline .github/workflows/black.yml:2]: The linters workflow will take care of the 'black' check. I do not think we need a separate workflow for black.

**Comment by GomathiselviS** [comment]
Should we change the amazon.aws version [here](https://github.com/ansible-collections/community.aws/blob/main/galaxy.yml#L11) to the one available in galaxy? I think this should resolve the failures we see in sanity and unit tests. I may be wrong !

**Review by tremble — APPROVED** [review]

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

File: .github/workflows/all_green_ckeck.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      - main
+      - 'stable-*'
+    tags:
+      - '*'
+
+jobs:
+  changelog-and-linters:
+    uses: ./.github/workflows/changelog_and_linters.yml  # use the callable changelog_and_linters job to run tests
+  linters:
+    uses: ./.github/workflows/linters.yml  # use the callable linters job to run tests
```
Comment (GomathiselviS) [inline .github/workflows/all_green_ckeck.yml:26]: This workflow is missing. This line has to be removed or linters.yml should be added.

**Comment by GomathiselviS** [comment]
https://github.com/ansible-collections/community.aws/pull/1815 should fix the unit test failures in Github Actions.

**Review by GomathiselviS — APPROVED** [review]

**Comment by GomathiselviS** [comment]
Can you recheck after rebasing? The latest test-requirements file is not picked up.

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Review by tremble — APPROVED** [review]

**Comment by alinabuzachis** [comment]
regate

**Comment by alinabuzachis** [comment]
regate

**Pull request merged**